OpenSim/Simulation/Test/testManager.cpp

```diff
@@ -136,3 +162,138 @@ void testStationCalcWithManager()
+        double finalHeight = 
+            initHeights[i] + initSpeeds[i]*duration - 0.5*g*duration*duration;
+        double finalSpeed = initSpeeds[i] - g*duration;
+        double sliderHeight = sliderCoord.getValue(state);
+        double sliderSpeed = sliderCoord.getSpeedValue(state);
+        cout << "Slider: t = " << state.getTime() << ", h = " << sliderHeight
+            << ", v = " << sliderSpeed << " | Eq: t = " << integFinalTimes[i]
+            << ", h = " << finalHeight << ", v = " << finalSpeed;
+
+        SimTK_TEST_EQ(state.getTime(), integFinalTimes[i]);
+        SimTK_TEST_EQ(sliderHeight, finalHeight);
+        SimTK_TEST_EQ(sliderSpeed, finalSpeed);
+
+        // Use Station to get the location, velocity & acceleration in ground.
+        stationHeight = myStation->getLocationInGround(state)[1];
+        stationSpeed = myStation->getVelocityInGround(state)[1];
+
+        cout << " | Station: t = " << state.getTime() << ", h = " 
+            << stationHeight << ", v = " << stationSpeed << endl;
+
+        // Compare Station values with equation values
+        SimTK_TEST_EQ(stationHeight, finalHeight);
+        SimTK_TEST_EQ(stationSpeed, finalSpeed);
+    }
+
+}
+
+void testExcitationUpdatesWithManager()
+{
+    cout << "Running testExcitationUpdatesWithManager" << endl;
+    LoadOpenSimLibrary("osimActuators");
+    Model arm("arm26.osim");
+
+    const Set<Muscle> &muscleSet = arm.getMuscles();
+    PrescribedController* controller = new PrescribedController();
+    controller->addActuator(muscleSet.get(0));
+    Constant* fn = new Constant(0);
+    controller->prescribeControlForActuator(0, fn);
+    arm.addController(controller);
+
+    SimTK::State& state = arm.initSystem();
+    state.setTime(0);
+    double stepsize = 0.01;
+    muscleSet.get(0).setActivation(state, 0.1);
+    for (int i = 0; i < 3; ++i)
+    {
+        arm.realizeDynamics(state);
+        double initAct = muscleSet.get(0).getActivation(state);
+        double excitation = 0.2 + i*0.2;
+
+        FunctionSet& fnset = controller->upd_ControlFunctions();
+        Constant* fn = dynamic_cast<Constant*>(&fnset[0]);
+        fn->setValue(excitation);
+
+        Manager manager(arm);
+        manager.initialize(state);
+        state = manager.integrate(stepsize*(i + 1));
+
+        arm.realizeDynamics(state);
+        double finalAct = muscleSet.get(0).getActivation(state);
+        double finalExcitation = muscleSet.get(0).getExcitation(state);
+        cout << state.getTime() << " " << excitation;
+        cout << " " << finalAct << endl;
+        // Check if excitation is correct
+        SimTK_TEST_EQ(excitation, finalExcitation);
+        // If excitation is not set correctly, it will go back to the default
+        // after the first integration. We can check if this is happening by
+        // ensuring the activation is increasing on each integration.
+        SimTK_TEST(finalAct > initAct);
```

Comment:
In line with the suggested changes, add `initAct = finalAct` to use to initialize the next integration loop.